app/controllers/products_controller.rb

```diff
@@ -2,12 +2,18 @@
 
 class ProductsController < ApplicationController
   before_action :product, only: %i[edit update show]
-  before_action :pagination, only: %i[index]
+  before_action :pagination, only: %i[admin_index index]
   skip_forgery_protection
 
   MAX_PAGE = (Product.count / 10.0).ceil
 
-  def index; end
+  def admin_index
+    render 'products/index'
+  end
+
+  def index
+    @cart_item = current_cart.cart_items.new
+  end
```

Comment:
let's keep this logic in one method, just add some condition to create cart_item only for non-admin users

Reply:
This two different methods for will_paginate, because otherwise it goes to the upper route in routes, and when I click to different page for example in user page - it can go to admin
https://stackoverflow.com/questions/57818084/how-can-overwrite-the-will-paginate-urls-will-paginate-gem-generate-wrong-url